src/plugins/embeddable/public/lib/embeddables/embeddable_root.tsx

```diff
@@ -0,0 +1,69 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import React from 'react';
+import { EuiLoadingSpinner } from '@elastic/eui';
+import { EuiText } from '@elastic/eui';
+import { IEmbeddable } from './i_embeddable';
+
+interface Props {
+  embeddable?: IEmbeddable;
+  loading?: boolean;
+  error?: string;
+}
+
+export class EmbeddableRoot extends React.Component<Props> {
```

Comment:
Couldn't `EmbeddableFactoryRenderer` and `EmbeddableRoot` be the same component?
It feels like they couldn't really exist one without the other, or could they?

Reply:
EmbeddableFactoryRenderer encapsulates more logic, but EmbeddableRoot if helpful if you need a reference to the embeddable object, like in `todo_embeddable_example`.  Adding some text there:

```

                You may also notice this example uses `EmbeddableRoot` instead of the
                `EmbeddableFactoryRenderer` helper component. This is because it needs a reference
                to the embeddable to update it, and `EmbeddableFactoryRenderer` creates and holds
                the embeddable instance internally.
```